### PR TITLE
KIALI-911 New endpoint for namespace health

### PR DESCRIPTION
**Pull request by jotak**
Some refactoring to share code with ServiceList health fetching (which also relies on namespace)
Also add tests for namespace metrics handler

### CODE REVIEW

File: services/business/health.go [hunk 1]
```diff
@@ -23,6 +29,75 @@ func (in *HealthService) GetServiceHealth(namespace, service, rateInterval strin
...
+	for _, item := range sl.Services.Items {
+		sPods := kubernetes.FilterPodsForService(&item, sl.Pods)
+		depls := kubernetes.FilterDeploymentsForService(&item, sPods, sl.Deployments)
+		statuses := castDeploymentsStatuses(depls)
+		allHealth[item.Name] = &models.Health{
+			DeploymentStatuses: statuses,
+			DeploymentsFetched: true}
+	}
+
+	// Fetch services requests rates
```
Comment (lucasponce) [inline services/business/health.go]: Perhaps the "Extract deployment statuses" and "Fetch services requests rates" blocks can be run concurrently

File: services/business/services.go [hunk 2]
```diff
@@ -28,61 +24,32 @@ func (in *SvcService) GetServiceList(namespace, rateInterval string) (*models.Se
 	if err != nil {
 		return nil, err
 	}
-
-	// Fetch services requests rates
-	inRates, outRates, _ := in.prom.GetNamespaceServicesRequestRates(namespace, rateInterval)
+	nsHealth := in.health.getNamespaceHealth(namespace, kubernetesServices, rateInterval)
```
Comment (lucasponce) [inline services/business/services.go]: I thought the goal was to separate the health from the GetServiceList to perform two async calls.
But here I see it's also fetched.
Perhaps I missed some detail.

File: services/business/health.go (hunk 1 above)
Comment (jotak) [reply services/business/health.go]: Yeah but "Extract deployment statuses" is I think super-fast (probably a matter of few milliseconds if not nanos); it doesn't fetch k8s api, it already has everything in `sl *kubernetes.ServiceList`.

File: services/business/services.go (hunk 2 above)
Comment (jotak) [reply services/business/services.go]: No, not for this task. Kiali-911 is about graph only. I think there's another one to change ServiceList behaviour

**Pull request merged**